WordPress/src/main/java/org/wordpress/android/ui/reader/actions/ReaderBlogActions.java

```diff
@@ -80,9 +80,17 @@ public void onResponse(JSONObject jsonObject) {
             public void onErrorResponse(VolleyError volleyError) {
                 AppLog.w(T.READER, "blog " + actionName + " failed with error");
                 AppLog.e(T.READER, volleyError);
-                localRevertFollowBlogId(blogId, isAskingToFollow);
-                if (actionListener != null) {
-                    actionListener.onActionResult(false);
+                // check if we get a 403 when unfollowing - this will happen when we attempt
+                // to unfollow a blog that no longer exists - the workaround is to unfollow
+                // by url - note that the v1.2 endpoint will return a 404 in this situation
+                int status = VolleyUtils.statusCodeFromVolleyError(volleyError);
+                if (status == 403 && !isAskingToFollow) {
+                    internalUnfollowBlogByUrl(blogId, actionListener);
```

Comment:
This is super minor, but it might be helpful to add a comment here stating that the `actionListener` methods will be called from within the method. At first glance, it looks no action is called after the request which of course not the case.